lib/modding/CModVersion.cpp

```diff
@@ -45,19 +45,49 @@ CModVersion CModVersion::fromString(std::string from)
 
 std::string CModVersion::toString() const
 {
-	return std::to_string(major) + '.' + std::to_string(minor) + '.' + std::to_string(patch);
+	std::string res;
+	if(major != Any)
+	{
+		res += std::to_string(major);
+		if(minor != Any)
+		{
+			res += '.' + std::to_string(minor);
+			if(patch != Any)
+				res += '.' + std::to_string(patch);
+		}
+	}
+	return res;
 }
 
 bool CModVersion::compatible(const CModVersion & other, bool checkMinor, bool checkPatch) const
 {
+	if(minor == Any || other.minor == Any)
+		checkMinor = false;
+	if(patch == Any || other.patch == Any)
+		checkPatch = false;
```

Comment:
Perhaps use separate variables instead of changing function input parameters? Should help a bit with code clarity.
```suggestion
bool doCheckMinor = checkMinor && minor != Any && other.minor != Any;
bool doCheckPatch = checkPatch && patch != Any && other.patch != Any;
```
(and obviously use these two variables below instead of function parameters)